service/history/conflictResolver.go

```diff
@@ -107,6 +108,9 @@ func (r *conflictResolver) reset(replayEventID int64, startTime time.Time) (*mut
 	// the last updated time is not important here, since this should be updated with event time afterwards
 	resetMutableStateBuilder.executionInfo.LastUpdatedTimestamp = startTime
 
+	sourceCluster := r.clusterMetadata.ClusterNameForFailoverVersion(resetMutableStateBuilder.GetCurrentVersion())
```

Comment:
I think you should read the version from last event rather than rely on CurrentVersion.